File: LeaseGuard/app/src/main/java/com/leaseguard/leaseguard/models/LeaseDocument.java
package com.leaseguard.leaseguard.models;

import java.util.UUID;

public class LeaseDocument {
    private String title;
    private String address;
    private int rent;
    private String dateRange;
    private int numIssues;
    private String uuid;

    public LeaseDocument(String title, String address, int rent, String dateRange, int numIssues) {
        this.title = title;
        this.address = address;
        this.rent = rent;
        this.dateRange = dateRange;
        this.numIssues = numIssues;
        this.uuid = UUID.randomUUID().toString();
    }

    public String getTitle() {
        return title;
    }

    public String getAddress() {
        return address;
    }

    public int getRent() {
        return rent;
    }

    public String getDateRange() {
        return dateRange;
    }

    public int getNumIssues() {
        return numIssues;
    }

    public String getUuid() { return uuid; }
}

File: LeaseGuard/app/src/main/java/com/leaseguard/leaseguard/models/LeaseDocument.kt
package com.leaseguard.leaseguard.models;

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * LeaseDocument model class used to represent an entity that will be stored in
 * our SQLite database within the table 'lease_table'
 */
@Entity(tableName = "lease_table")
class LeaseDocument(

        @PrimaryKey
        val id: String,
        val title: String, // Landlord / Property Management name
        val address: String,
        val rent: Int, // Rent amount
        val date: String, // Date range of lease
        val numIssues: Int,
        val issueDetails : String, // Rent Issue details stored as JSON text
        val status: String, // Completion status
        val thumbnail : ByteArray, // Thumbnail image
        val documentName : String

        )
